launchpod/src/test/java/com/google/launchpod/FormHandlerServletTest.java

```diff
@@ -0,0 +1,420 @@
+        for (Entity entity : results.asIterable()) {
+            thrown.expect(Exception.class);
+            thrown.expectMessage("There was no MP3 link inputted. Try again");
+        }
+    }
+
+    // This test can only be run if xmlString() is public.
+    @Test
+    public void generateXmlString() {
+        String xml = servlet.xmlString(TEST_PODCAST_TITLE, TEST_MP3_LINK, TEST_PUBDATE);
+        assertNotNull(xml);
+        assertEquals(xml, TEST_XML_STRING);
+    }
+
+    /**
+     * Asserts that doGet() returns correct URL when there's only one entity in Datastore period.
+     * @throws Exception
+    */
+    @Test
+    public void doGetReturnsOneUrlForOneUserOneEntity() throws Exception {
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+        Entity entity = makeEntity(TEST_PODCAST_TITLE, TEST_MP3_LINK, TEST_TIMESTAMP, TEST_XML_STRING, TEST_EMAIL);
+        ds.put(entity);
+        String id = KeyFactory.keyToString(entity.getKey());
+        
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter writer = new PrintWriter(stringWriter);
+        when(response.getWriter()).thenReturn(writer);
+
+        servlet.doGet(request, response);
+        
+        verify(response, atLeast(1)).setContentType("text/html");
+        writer.flush();
+        String testUrl = createUrl(TEST_PODCAST_TITLE, id);
+        assertEquals(stringWriter.toString(), testUrl); // only one URL in Datastore
+    }
+
+    /**
+     * Asserts that doGet() returns correct URLs when there are multiple entities in Datastore belonging to the same user.
+     * @throws Exception
+    */
+    @Test
+    public void doGetReturnsMultipleUrlsForOneUser() throws Exception {
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+        Entity entity = makeEntity(TEST_PODCAST_TITLE, TEST_MP3_LINK, TEST_TIMESTAMP, TEST_XML_STRING, TEST_EMAIL);
+        Entity entityTwo = makeEntity(TEST_PODCAST_TITLE, TEST_MP3_LINK, TEST_TIMESTAMP, TEST_XML_STRING, TEST_EMAIL);
+        String id = KeyFactory.keyToString(entity.getKey());
+        String idTwo = KeyFactory.keyToString(entityTwo.getKey());
+        ds.put(entity);
+        ds.put(entityTwo);
+        
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter writer = new PrintWriter(stringWriter);
+        when(response.getWriter()).thenReturn(writer);
+
+        servlet.doGet(request, response);
+        
+        verify(response, atLeast(1)).setContentType("text/html");
+        writer.flush();
+        String testUrls = createUrl(TEST_PODCAST_TITLE, id) + createUrl(TEST_PODCAST_TITLE, idTwo);
+        assertEquals(stringWriter.toString(), testUrls);
+    }
+
+    /**
+     * Asserts that doGet() returns the correct URL when there are multiple entities in Datastore belonging to different users.
+     * @throws Exception
+    */
+    @Test
+    public void doGetReturnsOneUrlForOneUserMultipleEntities() throws Exception {
```

Comment:
This seems to return multiple urls for multiple users, rather than one url for one user with multiple entities.

Reply:
Will rename this method!